atoma-service/src/middleware.rs

```diff
@@ -1674,14 +1674,14 @@ pub mod utils {
                         endpoint: endpoint.to_string(),
                     });
                 }
-                occupied_entry.remove();
```

Comment:
Why ? The ellapsed time has passed, so you should simply remove it.

Reply:
if you remove it here, then how do I know if you have to check for lower threshold? you don't so you applied to all, which is bad, the flag should be turned off when both time pass and lower threshold is hit